src/Utilities/Compiler/Options/SymbolNamesWithValueOption.cs

```diff
@@ -271,240 +270,73 @@ private bool TryGetFirstWildcardMatch(ISymbol symbol, out KeyValuePair<string, T
-                        currentType = currentType.ContainingType;
-                    }
-
-                    var currentNamespace = parameter.Type.ContainingNamespace;
-                    while (currentNamespace != null && !currentNamespace.IsGlobalNamespace)
-                    {
-                        builder.Insert(insertionPosition, currentNamespace.Name);
-                        builder.Insert(insertionPosition, ".");
-                        currentNamespace = currentNamespace.ContainingNamespace;
-                    }
-
-                    if (builder[insertionPosition] == '.')
-                    {
-                        builder.Remove(insertionPosition, 1);
-                    }
-                }
+                firstMatch = partialFirstMatchOrDefault;
+                _wildcardMatchResult.AddOrUpdate(symbol, firstMatch, (s, match) => partialFirstMatchOrDefault);
+                return true;
             }
 
-            static bool TryGetSymbolPartialMatch(StringBuilder builder, ISymbol symbol, string symbolName,
-                ImmutableDictionary<SymbolKind, ImmutableDictionary<string, TValue>> wildcardNamesBySymbolKind,
-                ConcurrentDictionary<ISymbol, KeyValuePair<string, TValue>> wildcardMatchResult,
-                out KeyValuePair<string, TValue> firstMatch)
-            {
-                if (builder.Length > 0 && builder[0] != '.')
-                {
-                    builder.Insert(0, ".");
-                }
-
-                builder.Insert(0, symbolName);
-
-                return TryGetFirstWildcardMatch(AllKinds, symbol, symbolName, wildcardNamesBySymbolKind, wildcardMatchResult, out firstMatch);
-            }
+            // Nothing was found
+            firstMatch = NoWildcardMatch;
+            _wildcardMatchResult.AddOrUpdate(symbol, firstMatch, (s, match) => NoWildcardMatch);
+            return false;
 
-            static bool TryGetFirstWildcardMatch(SymbolKind kind, ISymbol symbol, string symbolName,
-                ImmutableDictionary<SymbolKind, ImmutableDictionary<string, TValue>> wildcardNamesBySymbolKind,
-                ConcurrentDictionary<ISymbol, KeyValuePair<string, TValue>> wildcardMatchResult,
-                out KeyValuePair<string, TValue> firstMatch)
+            static string GetDeclarationId(ISymbol symbol)
             {
-                // Is there a cached result?
-                if (wildcardMatchResult.ContainsKey(symbol))
-                {
-                    firstMatch = wildcardMatchResult[symbol];
-                    return !firstMatch.Equals(NoWildcardMatch);
-                }
-
-                // Is there any wildcard entry for this symbol kind?
-                if (!wildcardNamesBySymbolKind.ContainsKey(kind))
-                {
-                    firstMatch = NoWildcardMatch;
-                    return false;
-                }
+                var declarationIdWithoutPrefix = DocumentationCommentId.CreateDeclarationId(symbol)[2..];
```

Comment:
I couldn't find any better way. We get called for each symbol walking up on the inheritance chain but we need the fully qualified part every time.